src/app/rundown-view/components/minishelf/mini-shelf.component.html

```diff
@@ -0,0 +1,20 @@
+<div
+  class="c-mini-shelf"
+  title="{{ segment.name }}"
+  (click)="executeAction()"
+>
+<!--  TODO: add ThumbnailComponent and use it here-->
+    <img
+        class="c-mini-shelf__thumbnail"
+        src="{{ mediaPreviewUrl }}"
+        alt="{{ segment.name? segment.name : 'Title is not available.' }}"
+        (error)=handleMissingImage($event)
+    />
+    <div class="c-mini-shelf__note">
+        <span class="c-mini-shelf__title"
+        >{{ getSanitizedTitle() ? getSanitizedTitle() : 'Title is not available.'  | slice:0:30 | uppercase }}{{ segment.name.length > 30 ? '...' : '' }}</span>
+        <span class="c-mini-shelf__duration">{{ mediaDuration === 0 || isNaN(mediaDuration) ? '-- : --' : mediaDuration | timer : 'HH?:mm:ss' }}</span>
```

Comment:
The logic here seems like something that could belong inside the `timer` pipe. So if the pipe receives an invalid input, it just returns `--:--`.

Reply:
Indeed, but there we'll find tests that contradict with this, with leads me in direction that the timer pipe contract is stated already on this subject, i.e. if duration is negative pipe will display '-', but if it is negative 500ms will return '01' and so on.

Reply:
As I see it, the issue lies somewhere else than here. First of all, why do we receive a `NaN`? And why won't we put 0 milliseconds into the timer pipe?